docs/src/resolve.md

```diff
@@ -0,0 +1,69 @@
+# [Resolve and in-place solve](@id resolve)
+
+It is very convenient to pre-allocate the memory used during the optimization of a given problem either from improved memory management or when re-solving the same or a similar problem.
+
+Let us consider the following 2-dimensional unconstrained problem
+```math
+\begin{aligned}
+\min \quad & f(x):= x₂ exp(x₁) 
+\end{aligned}
+```
+Using `JSOSuite` usual `minimize` function, the problem can be solved as follows
+```@example ex1
+using JSOSuite
+f(x) = x[2]^2 * exp(x[1]^2)
+x0 = ones(Float64, 2)
+stats = minimize(f, x0)
+```
+Using L-BFGS, the problem is locally solved.
+
+## In-place solve
+
+If we want to solve the same problem several time, for instance, for several initial guess it is recommended to use in-place solve.
+```@example ex1
+using ADNLPModels, JSOSolvers, SolverCore
+f(x) = x[2]^2 * exp(x[1]^2)
+x0 = ones(Float64, 2)
+nlp = ADNLPModel(f, x0) # or use JuMP
+solver = JSOSolvers.LBFGSSolver(nlp)
```

Comment:
Maybe explain that you can’t just pass `f` anymore, but you need to create a model.

Reply:
I added a paragraph above to justify further. That would be something great to add too, the `solve!(solver, f, x0, stats; kwargs...)`.